ape_hardhat/provider.py

```diff
@@ -296,16 +296,20 @@ def _set_web3(self):
 
         # Handle if using PoA Hardhat
         try:
-            block = self.web3.eth.get_block(0)
+            first_block = self.web3.eth.get_block(0)
+            latest_block = self.web3.eth.get_block("latest")
         except ExtraDataLengthError:
-            began_poa = True
+            contains_poa = True
         else:
-            began_poa = (
-                "proofOfAuthorityData" in block
-                or len(block.get("extraData", "")) > MAX_EXTRADATA_LENGTH
-            )
+            for block in (first_block, latest_block):
+                contains_poa = (
+                    "proofOfAuthorityData" in block
+                    or len(block.get("extraData", "")) > MAX_EXTRADATA_LENGTH
+                )
+                if contains_poa:
+                    break
 
-        if began_poa:
+        if contains_poa:
```

Comment:
Some notes:
1. `any` works with iterators
2. `any` short circuits on first positive match

Might look good as something like this:
```py
block_ids = iter([0, "latest"])
# add your detection logic against the block fetched via RPC, inside a function
blocks_with_poa = map(..., block_ids)

if any(blocks_with_poa):
   ...
```

This way you'll reduce one extra RPC call for a block in the off chance it matches on the first block

Reply:
https://github.com/ApeWorX/ape-hardhat/pull/134/files